src/main/java/jenkins/plugins/ui_samples/DropdownList.java

```diff
@@ -65,22 +65,16 @@ public void doSubmit(StaplerRequest req, StaplerResponse rsp) throws ServletExce
         }
     }
 
-    public static class FruitDescriptor extends Descriptor<Fruit> {
-        public FruitDescriptor(Class<? extends Fruit> clazz) {
-            super(clazz);
-        }
-        public String getDisplayName() {
-            return clazz.getSimpleName();
-        }
-    }
+    public static class FruitDescriptor extends Descriptor<Fruit> {}
 
     public static class Apple extends Fruit {
         private int seeds;
         @DataBoundConstructor public Apple(int seeds) {
             super("Apple");
             this.seeds = seeds;
         }
-        @Extension public static final FruitDescriptor D = new FruitDescriptor(Apple.class);
+        @Extension
+        public static final class DescriptorImpl extends FruitDescriptor {}
```

Comment:
Ah, this is even simpler than I thought, the Apple.class is derived automatically?

Reply:
Yes.